versioned_docs/version-2021sp/lecture3.md

```diff
@@ -43,4 +43,358 @@ From now on, we will be using a `tsconfig.json` file within every Node project w
 
 Use the link above to install Postman.
 
-The rest of the notes is coming soon!
+## Postman and Request Bodies
+
+### Postman
+
+Instead of always going to the endpoint in the browser, a robust way of testing
+our endpoints is to use Postman.
+
+Postman is a software that allows you to simulate requests that could be sent by
+a user to your backend. It is useful for testing and ensuring that the behavior
+of your requests (including necessary headers) is what you expect.
+
+### POST Request
+
+Usually when you want to send a `POST` request you also want to send information
+with it. Situationally, you want to do this using request bodies rather than
+query parameters.
+
+Let's say we have this `addSong` endpoint:
+
+```ts title="index.ts"
+app.post('/addSong', (req, res) => {
+  const song: songtype = { name: req.body.name, rating: req.body.rating };
+  console.log(song);
+  songs.push(song);
+  res.send(`Song ${req.body.name} added!`);
+});
+```
+
+where `songtype` is:
+
+```ts title="index.ts"
+type songtype = {
```

Comment:
maybe replace this instead with `Song` since that's better naming practice

Reply:
Good idea